modules/lib/lib-content/src/main/resources/lib/xp/content-graphql.js

```diff
@@ -0,0 +1,255 @@
+ * @param {string[]} [params.contentTypes] Content types to filter on.
+ *
+ * @returns {Object} Result of query.
+ */
+exports.query = function (params) {
+    var bean = __.newBean('com.enonic.xp.lib.content.QueryContentHandler');
+    bean.start = params.start;
+    bean.count = params.count;
+    bean.query = nullOrValue(params.query);
+    bean.sort = nullOrValue(params.sort);
+    bean.aggregations = __.toScriptValue(params.aggregations);
+    bean.contentTypes = __.toScriptValue(params.contentTypes);
+    bean.filters = __.toScriptValue(params.filters);
+    bean.highlight = __.toScriptValue(params.highlight);
+    return __.toNativeObjectIncludeNullValues(bean.execute());
+};
+
+/**
+ * Check if content exists.
+ *
+ * @example-ref examples/content/exists.js
+ *
+ * @param {string} [params.key] content id.
+ *
+ * @returns {boolean} True if exist, false otherwise.
+ */
+exports.exists = function (params) {
+    var bean = __.newBean('com.enonic.xp.lib.content.ContentExistsHandler');
+
+    bean.key = required(params, 'key');
+
+    return __.toNativeObjectIncludeNullValues(bean.execute());
+};
+
+/**
+ * Gets permissions on a content.
+ *
+ * @example-ref examples/content/getPermissions.js
+ *
+ * @param {object} params JSON parameters.
+ * @param {string} params.key Path or id of the content.
+ * @returns {object} Content permissions.
+ */
+exports.getPermissions = function (params) {
+    var bean = __.newBean('com.enonic.xp.lib.content.GetPermissionsHandler');
+
+    if (params.key) {
+        bean.key = params.key;
+    }
+    return __.toNativeObjectIncludeNullValues(bean.execute());
+};
+
+/**
+ * Returns the properties and icon of the specified content type.
+ *
+ * @example-ref examples/content/getType.js
+ *
+ * @param name Name of the content type, as 'app:name' (e.g. 'com.enonic.myapp:article').
+ * @returns {ContentType} The content type object if found, or null otherwise. See ContentType type definition below.
+ */
+exports.getType = function (name) {
+    var bean = __.newBean('com.enonic.xp.lib.content.ContentTypeHandler');
+    bean.name = nullOrValue(name);
+    return __.toNativeObjectIncludeNullValues(bean.getContentType());
+};
+
+/**
+ * Returns the list of all the content types currently registered in the system.
+ *
+ * @example-ref examples/content/getTypes.js
+ *
+ * @returns {ContentType[]} Array with all the content types found. See ContentType type definition below.
+ */
+exports.getTypes = function () {
+    var bean = __.newBean('com.enonic.xp.lib.content.ContentTypeHandler');
+    return __.toNativeObjectIncludeNullValues(bean.getAllContentTypes());
+};
+
+/**
+ * Returns outbound dependencies on a content.
+ *
+ * @param {object} params JSON parameters.
+ * @param {string} params.key Path or id of the content.
+ * @returns {object} Content Ids.
+ */
+exports.getOutboundDependencies = function (params) {
+    var bean = __.newBean('com.enonic.xp.lib.content.GetOutboundDependenciesHandler');
```

Comment:
Codacy found an issue: ['__' is not defined.](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=6266264)